tests/test_client.py

```diff
@@ -536,104 +647,115 @@ def test_index_create_with_duplicate_acl_authz(swg_index_client):
+def test_index_create_with_invalid_did(client, user):
     data = get_doc()
 
     data["did"] = "3d313755-cbb4-4b0fdfdfd8-899d-7bbac1f6e67dfdd"
-
-    with pytest.raises(ApiException) as e:
-        swg_index_client.add_entry(data)
-        assert e.status == 400
+    response = client.post("/index/", json=data, headers=user)
+    assert response.status_code == 400
 
 
-def test_index_create_with_prefix(swg_index_client):
+def test_index_create_with_prefix(client, user):
     data = get_doc()
     data["did"] = "cdis:3d313755-cbb4-4b08-899d-7bbac1f6e67d"
 
-    r = swg_index_client.add_entry(data)
-    assert r.did == "cdis:3d313755-cbb4-4b08-899d-7bbac1f6e67d"
+    response = client.post("/index/", json=data, headers=user)
+    record = response.json
+    assert record["did"] == "cdis:3d313755-cbb4-4b08-899d-7bbac1f6e67d"
 
 
-def test_index_create_with_duplicate_did(swg_index_client):
+def test_index_create_with_duplicate_did(client, user):
     data = get_doc()
     data["did"] = "3d313755-cbb4-4b08-899d-7bbac1f6e67d"
 
-    swg_index_client.add_entry(data)
-
-    with pytest.raises(ApiException) as e:
-        swg_index_client.add_entry(data)
-        assert e.status == 400
+    response = client.post("/index/", json=data, headers=user)
+    response = client.post("/index/", json=data, headers=user)
+    assert response.status_code == 400
 
 
-def test_index_create_with_file_name(swg_index_client):
+def test_index_create_with_file_name(client, user):
     data = get_doc()
     data["file_name"] = "abc"
 
-    r = swg_index_client.add_entry(data)
-    r = swg_index_client.get_entry(r.did)
-    assert r.file_name == "abc"
+    res = client.post("/index/", json=data, headers=user)
+    rec = res.json
+    res = client.get("/index/" + rec["did"])
+    rec = res.json
+    assert rec["file_name"] == "abc"
 
 
-def test_index_create_with_version(swg_index_client):
+def test_index_create_with_version(client, user):
     data = get_doc()
     data["version"] = "ver_123"
 
-    r = swg_index_client.add_entry(data)
-    r = swg_index_client.get_entry(r.did)
-    assert r.version == data["version"]
+    res = client.post("/index/", json=data, headers=user)
+    rec = res.json
+    res = client.get("/index/" + rec["did"])
+    rec = res.json
+    assert rec["version"] == data["version"]
 
 
-def test_index_create_blank_record(swg_index_client):
+def test_index_create_blank_record(client, user):
     doc = {"uploader": "uploader_123", "baseid": "baseid_123"}
 
-    r = swg_index_client.create_blank_entry(doc)
-    assert r.did
-    res = swg_index_client.get_entry(r.did)
-    assert res.acl == []
-    assert res.authz == []
-    assert res.urls_metadata == {}
-    assert res.size is None
-    assert res.version is None
-    assert res.urls_metadata == {}
+    res = client.post("/index/blank/", json=doc, headers=user)
+    rec = res.json
+    assert rec["did"]
+    res = client.get("/index/" + rec["did"])
+    rec = res.json
+    assert rec["acl"] == []
```

Comment:
could use `assert_blank` here

Reply:
Im going to change this test a little bit because I realized the baseid aspect wasn't really tested beyond the status code assertion, and it will also allow me to use assert_blank correctly.